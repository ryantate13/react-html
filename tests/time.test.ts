import {time} from '../src';

describe('time', () => {
    it('renders with children', () => {
        expect(time({}, 'TEST')).toMatchSnapshot();
    });

    it('renders with empty attributes', () => {
        expect(time({})).toMatchSnapshot();
    });

    it('renders without attributes', () => {
        expect(time()).toMatchSnapshot();
    });

    it('renders with accessKey="a,b,c,d"', () => {
        expect(time({accessKey: "a,b,c,d"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="on"', () => {
        expect(time({autoCapitalize: "on"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="off"', () => {
        expect(time({autoCapitalize: "off"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="none"', () => {
        expect(time({autoCapitalize: "none"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="sentences"', () => {
        expect(time({autoCapitalize: "sentences"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="words"', () => {
        expect(time({autoCapitalize: "words"})).toMatchSnapshot();
    });

    it('renders with autoCapitalize="characters"', () => {
        expect(time({autoCapitalize: "characters"})).toMatchSnapshot();
    });

    it('renders with className="foo bar baz"', () => {
        expect(time({className: "foo bar baz"})).toMatchSnapshot();
    });

    it('renders with contentEditable="true"', () => {
        expect(time({contentEditable: "true"})).toMatchSnapshot();
    });

    it('renders with contentEditable="false"', () => {
        expect(time({contentEditable: "false"})).toMatchSnapshot();
    });

    it('renders with dir="ltr"', () => {
        expect(time({dir: "ltr"})).toMatchSnapshot();
    });

    it('renders with dir="rtl"', () => {
        expect(time({dir: "rtl"})).toMatchSnapshot();
    });

    it('renders with dir="auto"', () => {
        expect(time({dir: "auto"})).toMatchSnapshot();
    });

    it('renders with draggable="true"', () => {
        expect(time({draggable: "true"})).toMatchSnapshot();
    });

    it('renders with draggable="false"', () => {
        expect(time({draggable: "false"})).toMatchSnapshot();
    });

    it('renders with hidden="true"', () => {
        expect(time({hidden: true})).toMatchSnapshot();
    });

    it('renders with hidden="false"', () => {
        expect(time({hidden: false})).toMatchSnapshot();
    });

    it('renders with id="TEST"', () => {
        expect(time({id: "TEST"})).toMatchSnapshot();
    });

    it('renders with inputMode="none"', () => {
        expect(time({inputMode: "none"})).toMatchSnapshot();
    });

    it('renders with inputMode="text"', () => {
        expect(time({inputMode: "text"})).toMatchSnapshot();
    });

    it('renders with inputMode="tel"', () => {
        expect(time({inputMode: "tel"})).toMatchSnapshot();
    });

    it('renders with inputMode="email"', () => {
        expect(time({inputMode: "email"})).toMatchSnapshot();
    });

    it('renders with inputMode="url"', () => {
        expect(time({inputMode: "url"})).toMatchSnapshot();
    });

    it('renders with inputMode="numeric"', () => {
        expect(time({inputMode: "numeric"})).toMatchSnapshot();
    });

    it('renders with inputMode="decimal"', () => {
        expect(time({inputMode: "decimal"})).toMatchSnapshot();
    });

    it('renders with inputMode="search"', () => {
        expect(time({inputMode: "search"})).toMatchSnapshot();
    });

    it('renders with is="foo"', () => {
        expect(time({is: "foo"})).toMatchSnapshot();
    });

    it('renders with itemID="https://google.com/"', () => {
        expect(time({itemID: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemProp="https://google.com/"', () => {
        expect(time({itemProp: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with itemRef="foo bar"', () => {
        expect(time({itemRef: "foo bar"})).toMatchSnapshot();
    });

    it('renders with itemScope="true"', () => {
        expect(time({itemScope: true})).toMatchSnapshot();
    });

    it('renders with itemScope="false"', () => {
        expect(time({itemScope: false})).toMatchSnapshot();
    });

    it('renders with itemType="https://google.com/"', () => {
        expect(time({itemType: "https://google.com/"})).toMatchSnapshot();
    });

    it('renders with lang="es-419"', () => {
        expect(time({lang: "es-419"})).toMatchSnapshot();
    });

    it('renders with lang=""', () => {
        expect(time({lang: ""})).toMatchSnapshot();
    });

    it('renders with slot="TEST"', () => {
        expect(time({slot: "TEST"})).toMatchSnapshot();
    });

    it('renders with spellCheck="true"', () => {
        expect(time({spellCheck: "true"})).toMatchSnapshot();
    });

    it('renders with spellCheck="false"', () => {
        expect(time({spellCheck: "false"})).toMatchSnapshot();
    });

    it('renders with style="[object Object]"', () => {
        expect(time({style: {"color":"#000000"}})).toMatchSnapshot();
    });

    it('renders with tabIndex="1"', () => {
        expect(time({tabIndex: 1})).toMatchSnapshot();
    });

    it('renders with title="TEST"', () => {
        expect(time({title: "TEST"})).toMatchSnapshot();
    });

    it('renders with translate="yes"', () => {
        expect(time({translate: "yes"})).toMatchSnapshot();
    });

    it('renders with translate="no"', () => {
        expect(time({translate: "no"})).toMatchSnapshot();
    });

    it('renders with dateTime="undefined"', () => {
        expect(time({dateTime: undefined})).toMatchSnapshot();
    });
});