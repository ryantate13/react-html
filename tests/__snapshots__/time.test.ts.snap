// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time renders arrays of children 1`] = `
<time>
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
  TEST
</time>
`;

exports[`time renders with accessKey="a,b,c,d" 1`] = `
<time
  accessKey="a,b,c,d"
/>
`;

exports[`time renders with autoCapitalize="characters" 1`] = `
<time
  autoCapitalize="characters"
/>
`;

exports[`time renders with autoCapitalize="none" 1`] = `
<time
  autoCapitalize="none"
/>
`;

exports[`time renders with autoCapitalize="off" 1`] = `
<time
  autoCapitalize="off"
/>
`;

exports[`time renders with autoCapitalize="on" 1`] = `
<time
  autoCapitalize="on"
/>
`;

exports[`time renders with autoCapitalize="sentences" 1`] = `
<time
  autoCapitalize="sentences"
/>
`;

exports[`time renders with autoCapitalize="words" 1`] = `
<time
  autoCapitalize="words"
/>
`;

exports[`time renders with children 1`] = `
<time>
  TEST
</time>
`;

exports[`time renders with className="foo bar baz" 1`] = `
<time
  className="foo bar baz"
/>
`;

exports[`time renders with contentEditable="false" 1`] = `
<time
  contentEditable="false"
/>
`;

exports[`time renders with contentEditable="true" 1`] = `
<time
  contentEditable="true"
/>
`;

exports[`time renders with dateTime="12:00" 1`] = `
<time
  dateTime="12:00"
/>
`;

exports[`time renders with dir="auto" 1`] = `
<time
  dir="auto"
/>
`;

exports[`time renders with dir="ltr" 1`] = `
<time
  dir="ltr"
/>
`;

exports[`time renders with dir="rtl" 1`] = `
<time
  dir="rtl"
/>
`;

exports[`time renders with draggable="false" 1`] = `
<time
  draggable="false"
/>
`;

exports[`time renders with draggable="true" 1`] = `
<time
  draggable="true"
/>
`;

exports[`time renders with empty attributes 1`] = `<time />`;

exports[`time renders with hidden=false 1`] = `
<time
  hidden={false}
/>
`;

exports[`time renders with hidden=true 1`] = `
<time
  hidden={true}
/>
`;

exports[`time renders with id="TEST" 1`] = `
<time
  id="TEST"
/>
`;

exports[`time renders with inputMode="decimal" 1`] = `
<time
  inputMode="decimal"
/>
`;

exports[`time renders with inputMode="email" 1`] = `
<time
  inputMode="email"
/>
`;

exports[`time renders with inputMode="none" 1`] = `
<time
  inputMode="none"
/>
`;

exports[`time renders with inputMode="numeric" 1`] = `
<time
  inputMode="numeric"
/>
`;

exports[`time renders with inputMode="search" 1`] = `
<time
  inputMode="search"
/>
`;

exports[`time renders with inputMode="tel" 1`] = `
<time
  inputMode="tel"
/>
`;

exports[`time renders with inputMode="text" 1`] = `
<time
  inputMode="text"
/>
`;

exports[`time renders with inputMode="url" 1`] = `
<time
  inputMode="url"
/>
`;

exports[`time renders with is="foo" 1`] = `
<time
  is="foo"
/>
`;

exports[`time renders with itemID="https://google.com/" 1`] = `
<time
  itemID="https://google.com/"
/>
`;

exports[`time renders with itemProp="https://google.com/" 1`] = `
<time
  itemProp="https://google.com/"
/>
`;

exports[`time renders with itemRef="foo bar" 1`] = `
<time
  itemRef="foo bar"
/>
`;

exports[`time renders with itemScope=false 1`] = `
<time
  itemScope={false}
/>
`;

exports[`time renders with itemScope=true 1`] = `
<time
  itemScope={true}
/>
`;

exports[`time renders with itemType="https://google.com/" 1`] = `
<time
  itemType="https://google.com/"
/>
`;

exports[`time renders with lang="" 1`] = `
<time
  lang=""
/>
`;

exports[`time renders with lang="es-419" 1`] = `
<time
  lang="es-419"
/>
`;

exports[`time renders with slot="TEST" 1`] = `
<time
  slot="TEST"
/>
`;

exports[`time renders with spellCheck="false" 1`] = `
<time
  spellCheck="false"
/>
`;

exports[`time renders with spellCheck="true" 1`] = `
<time
  spellCheck="true"
/>
`;

exports[`time renders with style={"color":"#000000"} 1`] = `
<time
  style={
    Object {
      "color": "#000000",
    }
  }
/>
`;

exports[`time renders with tabIndex=1 1`] = `
<time
  tabIndex={1}
/>
`;

exports[`time renders with title="TEST" 1`] = `
<time
  title="TEST"
/>
`;

exports[`time renders with translate="no" 1`] = `
<time
  translate="no"
/>
`;

exports[`time renders with translate="yes" 1`] = `
<time
  translate="yes"
/>
`;

exports[`time renders without attributes 1`] = `<time />`;
